### ISSUE TITLE
Problem when SwiftLint loads ".swiftlint.yml"

### ISSUE THREAD

**Issue by mendesbarreto** [realm/SwiftLint#514]
Hello every one!

I am having problem with SwiftLint at version 0.7.2 and I can't build my project on Xcode 7.2, when I include the  ".swiftlint.yml" file at the root of project, the follow message appears

```
Loading configuration from '.swiftlint.yml'
No lintable files found at path ''
Command /bin/sh failed with exit code 1
```

But build the project without the file  ".swiftlint.yml" run right, but I need specify some rules.

Any clues on what this issue might be?

Thanks for all!!!!

**Comment by jpsim** [comment]
Can you please post the contents of your `.swiftlint.yml` file?

**Comment by mendesbarreto** [comment]
Of course! 
```
disabled_rules: # rule identifiers to exclude from running
  - colon
  - comma
  - control_statement
opt_in_rules: # some rules are only opt-in
  - empty_count
  - missing_docs
  # Find all the available rules by running:
  # swiftlint rules
included: # paths to include during linting. `--path` is ignored if present.
  - Source
excluded: # paths to ignore during linting. Takes precedence over `included`.
  - Carthage
  - Pods
  - Source/ExcludedFolder
  - Source/ExcludedFile.swift

# configurable rules can be customized from this configuration file
# binary rules can set their severity level
force_cast: warning # implicitly
force_try:
  severity: warning # explicitly
# rules that have both warning and error levels, can set just the warning level
# implicitly
line_length: 110
# they can set both implicitly with an array
type_body_length:
  - 300 # warning
  - 400 # error
# or they can set both explicitly
file_length:
  warning: 500
  error: 1200
# naming rules can set warnings/errors for min_length and max_length
# additionally they can set excluded names
type_name:
  min_length: 4 # only warning
  max_length: # warning and error
    warning: 40
    error: 50
  excluded: iPhone # excluded via string
variable_name:
  min_length: # only min_length
    error: 4 # only error
  excluded: # excluded via string array
    - id
    - URL
    - GlobalAPIKey
reporter: "csv" # reporter type (xcode, json, csv, checkstyle)
```

I've try this version too and nothing
```
disabled_rules: # rule identifiers to exclude from running
  - colon
  - comma
  - control_statement
opt_in_rules: # some rules are only opt-in
  - empty_count
  - missing_docs
  # Find all the available rules by running:
  # swiftlint rules
included: # paths to include during linting. `--path` is ignored if present.
  - Source
excluded: # paths to ignore during linting. Takes precedence over `included`.
  - Carthage
  - Pods
  - Source/ExcludedFolder
  - Source/ExcludedFile.swift
```

Thanks for all

**Comment by jpsim** [comment]
What's the contents of your `Source` directory? Are there Swift files in `Source` other than in `ExcludedFolder` or `ExcludedFile.swift`?

**Comment by itsalan** [comment]
Reproduced this problem when a project has spaces in the name/path.  If the spaces are removed, it works fine.

**Comment by jpsim** [comment]
@itsalan could you share more complete steps to reproduce?.

**Comment by itsalan** [comment]
@jpsim Sure.  Create a new project, name it something like "Test Project".  In the .swiftlint.yml, add/replace these lines (or something similar referencing the path with spaces):  

`included: 
  - Test\ Project

excluded: 
   - Test\ Project/AppDelegate.swift`

SwiftLint will give you the error referenced above now.  It will run if the spaces are removed in the project path/folders, or if you remove the included/excluded configuration.

**Comment by jpsim** [comment]
That's because you shouldn't escape the string in your YAML file. YAML syntax already escapes spaces in strings, so it treats `\` as a literal character.

Change your YAML file to this:

```yaml
included:
  - Test Project
excluded:
  - Test Project/AppDelegate.swift
```

**Comment by jpsim** [comment]
To be very clear, I'm still waiting on steps to reproduce this issue. I can't trigger the behavior described here by following these steps:

```shell
$ tree -a
.
├── .swiftlint.yml
└── Test\ Project
    ├── AppDelegate.swift
    └── main.swift

1 directory, 3 files
$ cat .swiftlint.yml
included:
  - Test Project
excluded:
  - Test Project/AppDelegate.swift
$ cat "Test Project/AppDelegate.swift"
let a = 0
$ cat "Test Project/main.swift"
let b = 0
$ swiftlint
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'main.swift' (1/1)
Test Project/main.swift:1:1: error: Variable Name Violation: Variable name should be between 3 and 40 characters long: 'b' (variable_name)
Done linting! Found 1 violation, 1 serious in 1 file.
```

**Comment by NSCabezon** [comment]
I had the same problem and I don't know if it helps but I changed the .swiftlint.yml file to the equivalent of the "Test\ Project" folder and it started working. So from:

`├── .swiftlint.yml
└── Test\ Project
    ├── AppDelegate.swift
    └── main.swift`
to:
`└── Test\ Project
    ├── .swiftlint.yml
    ├── AppDelegate.swift
    └── main.swift`

(:

**Comment by jpsim** [comment]
@NSCabezon I'm sorry you're having trouble, but as indicated in a few comments in this thread, we have yet to get steps to reproduce. Care to share?

**Comment by AppsTitude** [comment]
I had the same issue but I solved it by using quotes: "Abc\ def". Don't forget the quotes!

**Comment by jpsim** [comment]
Closing as we never got steps to reproduce this.

**Comment by zontag** [comment]
Commenting ```- Source``` solve this issue here.

**Comment by hk05** [comment]
I just updated to xcode 10.2 and surprisingly ran into this problem, even with the default config file.

What helped for me was to comment out the Sources directory in the config file, like so:

included: # paths to include during linting. `--path` is ignored if present.
#  - Source

**Comment by yoKurt94** [comment]
I installed SwiftLint on my MacBook M1 via homebrew. I added the file path in the build phase as mentioned in the readme and SwiftLint works. However, if I add my `.swiftlint.yml` file, I get the error 
```
Linting Swift files in current working directory
Error: No lintable files found at paths: ''
```
It doesn't matter what I add into the "included" section, SwiftLint always complains that there is nothing `at paths: ''`. This looks to me as if SwiftLint doesn't recognise any directory I add?

**Comment by SimplyDanny** [comment]
Could you open a new issue explaining your folder structure, the script running SwiftLint, the `swiftlint.yml` configuration file, ...

Else your problem might get lost in here.

**Comment by yoKurt94** [comment]
I just uninstalled and reinstalled it via brew (with still the same .swiftlint.yml file in place) and now it works.
